# Online Dissertation Research Notebook

## Author: April Dawn Garrett
#### Affiliation: University of Vermont, Biology Department, Pespeni Lab 

## Projects title/descriptions

## Date started: 03/25/2018 
## Date end:  


# Table of contents (for 200 entries)    
* [Page 1:   Start: 2018-04-04](#id-section1). Urchin Capseq MS BWA Mem ReMapping Notes
* [Page 2:  ](#id-section2). Carbonic Anhydrase: Getting Allele Freqs From Files Melissa Mapped
* [Page 3:   Start: 2018-04-17](#id-section3). Processing OASV2 RNA Samples
* [Page 4:   Start: 2018-05-02](#id-section4). Working with vcf files (NoDups & with dups) to get AFs & compare to OASV2 DNA (in progress)
* [Page 5:  ](#id-section5).
* [Page 6:  ](#id-section6).
* [Page 7:  ](#id-section7).
* [Page 8:  ](#id-section8).
* [Page 9:  ](#id-section9).
* [Page 10:  ](#id-section10).
* [Page 11:  ](#id-section11).
* [Page 12:  ](#id-section12).
* [Page 13:  ](#id-section13).
* [Page 14:  ](#id-section14).
* [Page 15:  ](#id-section15).
* [Page 16:  ](#id-section16).
* [Page 17:  ](#id-section17).
* [Page 18:  ](#id-section18).
* [Page 19:  ](#id-section19).
* [Page 20:  ](#id-section20).
* [Page 21:  ](#id-section21).
* [Page 22:  ](#id-section22).
* [Page 23:  ](#id-section23).
* [Page 24:  ](#id-section24).
* [Page 25:  ](#id-section25).
* [Page 26:  ](#id-section26).
* [Page 27:  ](#id-section27).
* [Page 28:  ](#id-section28).
* [Page 29:  ](#id-section29).
* [Page 30:  ](#id-section30).
* [Page 31:  ](#id-section31).
* [Page 32:  ](#id-section32).
* [Page 33:  ](#id-section33).
* [Page 34:  ](#id-section34).
* [Page 35:  ](#id-section35).
* [Page 36:  ](#id-section36).
* [Page 37:  ](#id-section37).
* [Page 38:  ](#id-section38).
* [Page 39:  ](#id-section39).
* [Page 40:  ](#id-section40).
* [Page 41:  ](#id-section41).
* [Page 42:  ](#id-section42).
* [Page 43:  ](#id-section43).
* [Page 44:  ](#id-section44).
* [Page 45:  ](#id-section45).
* [Page 46:  ](#id-section46).
* [Page 47:  ](#id-section47).
* [Page 48:  ](#id-section48).
* [Page 49:  ](#id-section49).
* [Page 50:  ](#id-section50).
* [Page 51:  ](#id-section51).
* [Page 52:  ](#id-section52).
* [Page 53:  ](#id-section53).
* [Page 54:  ](#id-section54).
* [Page 55:  ](#id-section55).
* [Page 56:  ](#id-section56).
* [Page 57:  ](#id-section57).
* [Page 58:  ](#id-section58).
* [Page 59:  ](#id-section59).
* [Page 60:  ](#id-section60).
* [Page 61:  ](#id-section61).
* [Page 62:  ](#id-section62).
* [Page 63:  ](#id-section63).
* [Page 64:  ](#id-section64).
* [Page 65:  ](#id-section65).
* [Page 66:  ](#id-section66).
* [Page 67:  ](#id-section67).
* [Page 68:  ](#id-section68).
* [Page 69:  ](#id-section69).
* [Page 70:  ](#id-section70).
* [Page 71:  ](#id-section71).
* [Page 72:  ](#id-section72).
* [Page 73:  ](#id-section73).
* [Page 74:  ](#id-section74).
* [Page 75:  ](#id-section75).
* [Page 76:  ](#id-section76).
* [Page 77:  ](#id-section77).
* [Page 78:  ](#id-section78).
* [Page 79:  ](#id-section79).
* [Page 80:  ](#id-section80).
* [Page 81:  ](#id-section81).
* [Page 82:  ](#id-section82).
* [Page 83:  ](#id-section83).
* [Page 84:  ](#id-section84).
* [Page 85:  ](#id-section85).
* [Page 86:  ](#id-section86).
* [Page 87:  ](#id-section87).
* [Page 88:  ](#id-section88).
* [Page 89:  ](#id-section89).
* [Page 90:  ](#id-section90).
* [Page 91:  ](#id-section91).
* [Page 92:  ](#id-section92).
* [Page 93:  ](#id-section93).
* [Page 94:  ](#id-section94).
* [Page 95:  ](#id-section95).
* [Page 96:  ](#id-section96).
* [Page 97:  ](#id-section97).
* [Page 98:  ](#id-section98).
* [Page 99:  ](#id-section99).
* [Page 100:  ](#id-section100).
* [Page 101:  ](#id-section101).
* [Page 102:  ](#id-section102).
* [Page 103:  ](#id-section103).
* [Page 104:  ](#id-section104).
* [Page 105:  ](#id-section105).
* [Page 106:  ](#id-section106).
* [Page 107:  ](#id-section107).
* [Page 108:  ](#id-section108).
* [Page 109:  ](#id-section109).
* [Page 110:  ](#id-section110).
* [Page 111:  ](#id-section111).
* [Page 112:  ](#id-section112).
* [Page 113:  ](#id-section113).
* [Page 114:  ](#id-section114).
* [Page 115:  ](#id-section115).
* [Page 116:  ](#id-section116).
* [Page 117:  ](#id-section117).
* [Page 118:  ](#id-section118).
* [Page 119:  ](#id-section119).
* [Page 120:  ](#id-section120).
* [Page 121:  ](#id-section121).
* [Page 122:  ](#id-section122).
* [Page 123:  ](#id-section123).
* [Page 124:  ](#id-section124).
* [Page 125:  ](#id-section125).
* [Page 126:  ](#id-section126).
* [Page 127:  ](#id-section127).
* [Page 128:  ](#id-section128).
* [Page 129:  ](#id-section129).
* [Page 130:  ](#id-section130).
* [Page 131:  ](#id-section131).
* [Page 132:  ](#id-section132).
* [Page 133:  ](#id-section133).
* [Page 134:  ](#id-section134).
* [Page 135:  ](#id-section135).
* [Page 136:  ](#id-section136).
* [Page 137:  ](#id-section137).
* [Page 138:  ](#id-section138).
* [Page 139:  ](#id-section139).
* [Page 140:  ](#id-section140).
* [Page 141:  ](#id-section141).
* [Page 142:  ](#id-section142).
* [Page 143:  ](#id-section143).
* [Page 144:  ](#id-section144).
* [Page 145:  ](#id-section145).
* [Page 146:  ](#id-section146).
* [Page 147:  ](#id-section147).
* [Page 148:  ](#id-section148).
* [Page 149:  ](#id-section149).
* [Page 150:  ](#id-section150).
* [Page 151:  ](#id-section151).
* [Page 152:  ](#id-section152).
* [Page 153:  ](#id-section153).
* [Page 154:  ](#id-section154).
* [Page 155:  ](#id-section155).
* [Page 156:  ](#id-section156).
* [Page 157:  ](#id-section157).
* [Page 158:  ](#id-section158).
* [Page 159:  ](#id-section159).
* [Page 160:  ](#id-section160).
* [Page 161:  ](#id-section161).
* [Page 162:  ](#id-section162).
* [Page 163:  ](#id-section163).
* [Page 164:  ](#id-section164).
* [Page 165:  ](#id-section165).
* [Page 166:  ](#id-section166).
* [Page 167:  ](#id-section167).
* [Page 168:  ](#id-section168).
* [Page 169:  ](#id-section169).
* [Page 170:  ](#id-section170).
* [Page 171:  ](#id-section171).
* [Page 172:  ](#id-section172).
* [Page 173:  ](#id-section173).
* [Page 174:  ](#id-section174).
* [Page 175:  ](#id-section175).
* [Page 176:  ](#id-section176).
* [Page 177:  ](#id-section177).
* [Page 178:  ](#id-section178).
* [Page 179:  ](#id-section179).
* [Page 180:  ](#id-section180).
* [Page 181:  ](#id-section181).
* [Page 182:  ](#id-section182).
* [Page 183:  ](#id-section183).
* [Page 184:  ](#id-section184).
* [Page 185:  ](#id-section185).
* [Page 186:  ](#id-section186).
* [Page 187:  ](#id-section187).
* [Page 188:  ](#id-section188).
* [Page 189:  ](#id-section189).
* [Page 190:  ](#id-section190).
* [Page 191:  ](#id-section191).
* [Page 192:  ](#id-section192).
* [Page 193:  ](#id-section193).
* [Page 194:  ](#id-section194).
* [Page 195:  ](#id-section195).
* [Page 196:  ](#id-section196).
* [Page 197:  ](#id-section197).
* [Page 198:  ](#id-section198).
* [Page 199:  ](#id-section199).
* [Page 200:  ](#id-section200).


------

<div id='id-section1'/>    
    
### Page 1:  Urchin Capseq MS BWA Mem ReMapping Notes     


### April 4, 2018: Carbonic Anhydrase work. Adding read groups to bam files via picard tools and remaking vcf file for Melissa

java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=CC_SH_H_D1_0385.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/CC_SH_H_D1_0385.fastq.WithRG.bam \
      RGID=CC_SH_H_D1_0385 \
      RGLB=CC_SH_H_D1_0385 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=CC_SH_H_D1_0385
      
# Check to see that it read group was added     
/data/programs/samtools view -H CC_SH_H_D1_0385.fastq.WithRG.bam | grep "^@RG" > RG_CC_SH_H_D1_0385.txt

# Now do for all files (BELOW)

java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=CC_SH_H_D7_1000.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/CC_SH_H_D7_1000.fastq.WithRG.bam \
      RGID=CC_SH_H_D7_1000 \
      RGLB=CC_SH_H_D7_1000 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=CC_SH_H_D7_1000
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=CC_TP_L_D1_0385.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/CC_TP_L_D1_0385.fastq.WithRG.bam \
      RGID=CC_TP_L_D1_0385 \
      RGLB=CC_TP_L_D1_0385 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=CC_TP_L_D1_0385
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=CC_TP_L_D7_1000.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/CC_TP_L_D7_1000.fastq.WithRG.bam \
      RGID=CC_TP_L_D7_1000 \
      RGLB=CC_TP_L_D7_1000 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=CC_TP_L_D7_1000
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=NC_BD_L_D1_0385.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/NC_BD_L_D1_0385.fastq.WithRG.bam \
      RGID=NC_BD_L_D1_0385 \
      RGLB=NC_BD_L_D1_0385 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=NC_BD_L_D1_0385
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=NC_BD_L_D7_1000.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/NC_BD_L_D7_1000.fastq.WithRG.bam \
      RGID=NC_BD_L_D7_1000 \
      RGLB=NC_BD_L_D7_1000 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=NC_BD_L_D7_1000
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=NC_VD_H_D1_0385.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/NC_VD_H_D1_0385.fastq.WithRG.bam \
      RGID=NC_VD_H_D1_0385 \
      RGLB=NC_VD_H_D1_0385 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=NC_VD_H_D1_0385
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=NC_VD_H_D7_1000.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/NC_VD_H_D7_1000.fastq.WithRG.bam \
      RGID=NC_VD_H_D7_1000 \
      RGLB=NC_VD_H_D7_1000 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=NC_VD_H_D7_1000
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=OR_FC_L_D1_0385.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/OR_FC_L_D1_0385.fastq.WithRG.bam \
      RGID=OR_FC_L_D1_0385 \
      RGLB=OR_FC_L_D1_0385 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=OR_FC_L_D1_0385
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=OR_FC_L_D7_1000.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/OR_FC_L_D7_1000.fastq.WithRG.bam \
      RGID=OR_FC_L_D7_1000 \
      RGLB=OR_FC_L_D7_1000 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=OR_FC_L_D7_1000
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=OR_ST_H_D1_0385.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/OR_ST_H_D1_0385.fastq.WithRG.bam \
      RGID=OR_ST_H_D1_0385 \
      RGLB=OR_ST_H_D1_0385 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=OR_ST_H_D1_0385
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=OR_ST_H_D7_1002.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/OR_ST_H_D7_1002.fastq.WithRG.bam \
      RGID=OR_ST_H_D7_1002 \
      RGLB=OR_ST_H_D7_1002 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=OR_ST_H_D7_1002
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=SB_AL_L_D1_0385.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/SB_AL_L_D1_0385.fastq.WithRG.bam \
      RGID=SB_AL_L_D1_0385 \
      RGLB=SB_AL_L_D1_0385 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=SB_AL_L_D1_0385
      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=SB_AL_L_D7_1000.fastq.gz.bam \
      O=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/SB_AL_L_D7_1000.fastq.WithRG.bam \
      RGID=SB_AL_L_D7_1000 \
      RGLB=SB_AL_L_D7_1000 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=SB_AL_L_D7_1000
      
#######################################################################################################################
      
### MAKING VCF File



# Merge all the bam files into one
/data/programs/samtools merge merged.bam *.bam

## Making bash script for sorting and indexing merged bam file and making vcf file

#!/bin/bash

(comment out) NOTE: Need to specify a different tmp directory or it will fill the rhel-root...
/data/programs/sambamba_v0.6.0 sort -m 8G -t 8 -p --tmpdir=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/data/tmp merged.bam merged.sorted.bam

# Remove PCR duplicates (same as Picard tools)

/data/programs/sambamba_v0.6.0 markdup -t 8 -r --tmpdir=/users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/data/tmp merged.sorted.bam merged.sorted.rmdup.bam

# Run samtools index (with original samtools, not version 019)

/data/programs/samtools index merged.sorted.rmdup.bam

# Making VCF file with samtools version 1.4 # Removing skip-indels
/data/programs/samtools-1.4.1/samtools mpileup -u -t DP,AD,INFO/AD -f /data/urchinadult/carbonic_anhydrase_nt.fa merged.sorted.rmdup.bam | /data/programs/bcftools/bin/bcftools bcftools call -mv  > CarbonicAnhydrase_urchinlarvae2013.vcf

#######################################################################################################################

------

<div id='id-section3'/>  

### Page 3:  Processing OASV2 RNA Samples

### Start: April 17, 2018: OASV2 RNA Samples: FastQC/Trimmomatic

SAMPLES:
 D107_TAK_S17_L003_R1_001.fastq.gz
 D722_TAK_S16_L003_R1_001.fastq.gz
 D722_TruSeq_S18_L003_R1_001.fastq.gz
 D723_TAK_S15_L003_R1_001.fastq.gz

## LOOKING AT RAW FASTQ FILE QUALITY WITH FASTQC

/data/programs/FastQC/fastqc *.fastq.gz -o ./fastqc/raw

scp *.html amakukho@zoo.uvm.edu:~/public_html


## CLEANING READS (April 23, 2018)

#!/bin/bash  

for f1 in *_R1_001.fastq.gz

do

java -jar /data/programs/Trimmomatic-0.36/trimmomatic-0.36.jar SE \
        -threads 10 \
        -phred33 \
         "$f1" \
         /data/urchinlarvae2018/fastqc/cleaned/"$f1"_left_clean_unpaired.fq \
        ILLUMINACLIP:/data/urchinlarvae2018/fastqc/OASV2_RNAAdapters.fa:2:30:10 \
        SLIDINGWINDOW:20:2 \
        LEADING:2 \
        TRAILING:2 \
        MINLEN:35

done


## CHECKING NUMBER OF READS BEFORE/AFTER CLEANING (April 24, 2018):

echo $(cat yourfile.fastq|wc -l)/4|bc

# BEFORE:
D107_TAK_S17_L003_R1_001.fastq.gz -->  43871780
echo $(zcat D107_TAK_S17_L003_R1_001.fastq.gz|wc -l)/4|bc

D722_TAK_S16_L003_R1_001.fastq.gz  --> 48627152
echo $(zcat D722_TAK_S16_L003_R1_001.fastq.gz|wc -l)/4|bc

D722_TruSeq_S18_L003_R1_001.fastq.gz --> 59064939
echo $(zcat D722_TruSeq_S18_L003_R1_001.fastq.gz|wc -l)/4|bc

D723_TAK_S15_L003_R1_001.fastq.gz --> 36001001
echo $(zcat D723_TAK_S15_L003_R1_001.fastq.gz|wc -l)/4|bc


# AFTER:
D107_TAK_S17_L003_R1_001.fastq.gz_left_clean_unpaired.fq --> 42767363
echo $(cat D107_TAK_S17_L003_R1_001.fastq.gz_left_clean_unpaired.fq|wc -l)/4|bc

D722_TAK_S16_L003_R1_001.fastq.gz_left_clean_unpaired.fq --> 45155896
echo $(cat D722_TAK_S16_L003_R1_001.fastq.gz_left_clean_unpaired.fq|wc -l)/4|bc

D722_TruSeq_S18_L003_R1_001.fastq.gz_left_clean_unpaired.fq --> 59060167
echo $(cat D722_TruSeq_S18_L003_R1_001.fastq.gz_left_clean_unpaired.fq|wc -l)/4|bc

D723_TAK_S15_L003_R1_001.fastq.gz_left_clean_unpaired.fq --> 31817008
echo $(cat D723_TAK_S15_L003_R1_001.fastq.gz_left_clean_unpaired.fq|wc -l)/4|bc


# PERCENT LOSS OF READS:
D107: 2.5%
D722_TAK: 7.1%
D722_TruSeq: 0.008%
D723: 11.6%

# CHECKING CLEANED READS WITH FASTQC IN /data/urchinlarvae2018/fastqc/cleaned DIRECTORY:
/data/programs/FastQC/fastqc *.fq

moved fastqc files to new directory: /data/urchinlarvae2018/fastqc/cleaned/fastqc

scp *.html amakukho@zoo.uvm.edu:~/public_html

################################################################################################################################


### Start: April 24, 2018: OASV2 RNA Samples: MAPPING WITH BWAMEM

SCREEN ID: 9782.pts-1.pespenilab
/data/programs/bwa/bwa mem -t 4 -k 5 -R '@RG\tID:OASV2_RNA_D1_8_0_S_07\tSM:OASV2_RNA_D1_8_0_S_07\tPL:Illumina' ~/RESEARCH/OASV2/data/bwa_mapped/ref/ref /data/urchinlarvae2018/fastqc/cleaned/D107_TAK_S17_L003_R1_001.fastq.gz_left_clean_unpaired.fq > OASV2_RNA_D1_8_0_S_07_bwamem.sam

SCREEN ID: 9917.pts-1.pespenilab
/data/programs/bwa/bwa mem -t 4 -k 5 -R '@RG\tID:OASV2_RNA_D7_7_5_S_22\tSM:OASV2_RNA_D7_7_5_S_22\tPL:Illumina' ~/RESEARCH/OASV2/data/bwa_mapped/ref/ref /data/urchinlarvae2018/fastqc/cleaned/D722_TAK_S16_L003_R1_001.fastq.gz_left_clean_unpaired.fq > OASV2_RNA_D7_7_5_S_22_bwamem.sam

SCREEN ID: 9929.pts-1.pespenilab
/data/programs/bwa/bwa mem -t 4 -k 5 -R '@RG\tID:OASV2_RNA_D7_7_5_S_22_TruSeq\tSM:OASV2_RNA_D7_7_5_S_22_TruSeq\tPL:Illumina' ~/RESEARCH/OASV2/data/bwa_mapped/ref/ref /data/urchinlarvae2018/fastqc/cleaned/D722_TruSeq_S18_L003_R1_001.fastq.gz_left_clean_unpaired.fq > OASV2_RNA_D7_7_5_S_22_TruSeq_bwamem.sam

SCREEN ID: 9944.pts-1.pespenilab
/data/programs/bwa/bwa mem -t 4 -k 5 -R '@RG\tID:OASV2_RNA_D7_7_5_S_23\tSM:OASV2_RNA_D7_7_5_S_23\tPL:Illumina' ~/RESEARCH/OASV2/data/bwa_mapped/ref/ref /data/urchinlarvae2018/fastqc/cleaned/D723_TAK_S15_L003_R1_001.fastq.gz_left_clean_unpaired.fq > OASV2_RNA_D7_7_5_S_23_bwamem.sam

## SAM FILES FOUND IN: /data/urchinlarvae2018/mapped

################################################################################################################################
### Start: April 26, 2018: OASV2 RNA Samples: MAPPING WITH BWAMEM - SAM TO BAM AND SAMTOOLS FLAGSTAT

## CONVERTING SAM FILES TO BAM

# Making script called Bam2Sam.sh and putting in /data/urchinlarvae2018/scripts

#!/bin/bash

cd /data/urchinlarvae2018/mapped

for file in *.sam

do

/data/programs/samtools view -@ 16 -bhS "$file" >"$file.bam"

done

# Run chmod +x to make executable

## Ran Bam2Sam.sh in screen ID: 6885.pts-0.pespenilab


## CHECKING ON BAM FILES WITH SAMTOOLS FLAGSTAT TO GET % READS MAPPED
# Use /data/programs/samtools flagstat to check after converting sam to bam (before merging and sorting)

-bash-4.2$ /data/programs/samtools flagstat OASV2_RNA_D1_8_0_S_07_bwamem.sam.bam
44624556 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
1857193 + 0 supplementary
0 + 0 duplicates
39221596 + 0 mapped (87.89%:-nan%)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (-nan%:-nan%)
0 + 0 with itself and mate mapped
0 + 0 singletons (-nan%:-nan%)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)

-bash-4.2$ /data/programs/samtools flagstat OASV2_RNA_D7_7_5_S_22_bwamem.sam.bam
49420953 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
4265057 + 0 supplementary
0 + 0 duplicates
45575885 + 0 mapped (92.22%:-nan%)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (-nan%:-nan%)
0 + 0 with itself and mate mapped
0 + 0 singletons (-nan%:-nan%)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)

-bash-4.2$ /data/programs/samtools flagstat OASV2_RNA_D7_7_5_S_22_TruSeq_bwamem.sam.bam
65014018 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
5953851 + 0 supplementary
0 + 0 duplicates
57966089 + 0 mapped (89.16%:-nan%)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (-nan%:-nan%)
0 + 0 with itself and mate mapped
0 + 0 singletons (-nan%:-nan%)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)

-bash-4.2$ /data/programs/samtools flagstat OASV2_RNA_D7_7_5_S_23_bwamem.sam.bam
35058644 + 0 in total (QC-passed reads + QC-failed reads)
0 + 0 secondary
3241636 + 0 supplementary
0 + 0 duplicates
32550729 + 0 mapped (92.85%:-nan%)
0 + 0 paired in sequencing
0 + 0 read1
0 + 0 read2
0 + 0 properly paired (-nan%:-nan%)
0 + 0 with itself and mate mapped
0 + 0 singletons (-nan%:-nan%)
0 + 0 with mate mapped to a different chr
0 + 0 with mate mapped to a different chr (mapQ>=5)

################################################################################################################################
### Start: April 26, 2018: OASV2 RNA Samples: MERGING, SORTING, REMOVING DUPLICATES, AND INDEXING

# MERGING
# Merge all the bam files into one : Screen ID: 27053.pts-0.pespenilab
/data/programs/samtools merge merged.bam *.bam 

# SORTING
# NOTE: Need to specify a different tmp directory or it will fill the rhel-root... (CURRENTLY HERE)
/data/programs/sambamba_v0.6.0 sort -m 8G -t 8 -p --tmpdir=/data/urchinlarvae2018/mapped/tmp merged.bam merged.sorted.bam
SCREEN ID: 56281.pts-5.pespenilab


# REMOVING PCR DUPLICATES
# Remove PCR duplicates (same as Picard tools) # run in my directory /users/a/m/amakukho/RESEARCH/CarbonicAnhydrase/data
/data/programs/sambamba_v0.6.0 markdup -t 8 -r --tmpdir=/data/urchinlarvae2018/mapped/tmp merged.sorted.bam merged.sorted.rmdup.bam




################################################################################################################################
### Start: April 30, 2018: OASV2 RNA Samples: MERGING, SORTING, REMOVING DUPLICATES (PICARD TOOLS), AND INDEXING

## Keep getting the following error: "More than 16383 reference sequences are unsupported sambamba-markdup". Going to try using Picard Tools to mark duplicates. 


EXAMPLE:
 java ­jar MarkDuplicates.jar INPUT=$bamfile
       OUTPUT=$bamfile_dupFree.bam  REMOVE_DUPLICATES=true
       METRICS_FILE=metrics CREATE_INDEX=true
       VALIDATION_STRINGENCY=LENIENT


## NOTE: LB needs to be part of read group in order for read group to be considered in Picard Tools Mark Duplicates (http://seqanswers.com/forums/showthread.php?t=19540) (https://software.broadinstitute.org/gatk/documentation/article.php?id=6472), so need to replace the read groups via Picard Tools first:


java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=OASV2_RNA_D1_8_0_S_07_bwamem.sam.bam \
      O=OASV2_RNA_D1_8_0_S_07_bwamem.WithLB.bam \
      RGID=OASV2_RNA_D1_8_0_S_07 \
      RGLB=OASV2_RNA_D1_8_0_S_07 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=OASV2_RNA_D1_8_0_S_07     
# SCREEN ID: 8818.pts-13.pespenilab
      
      
       AddOrReplaceReadGroups \
      I=OASV2_RNA_D7_7_5_S_22_bwamem.sam.bam \
      O=OASV2_RNA_D7_7_5_S_22_bwamem.WithLB.bam \
      RGID=OASV2_RNA_D7_7_5_S_22 \
      RGLB=OASV2_RNA_D7_7_5_S_22 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=OASV2_RNA_D7_7_5_S_22
# SCREEN ID: 9346.pts-13.pespenilab



java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=OASV2_RNA_D7_7_5_S_22_TruSeq_bwamem.sam.bam \
      O=OASV2_RNA_D7_7_5_S_22_TruSeq_bwamem.WithLB.bam \
      RGID=OASV2_RNA_D7_7_5_S_22_TruSeq \
      RGLB=OASV2_RNA_D7_7_5_S_22_TruSeq \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=OASV2_RNA_D7_7_5_S_22_TruSeq   
# SCREEN ID: 9513.pts-13.pespenilab

      
java -jar /data/programs/picard.jar AddOrReplaceReadGroups \
      I=OASV2_RNA_D7_7_5_S_23_bwamem.sam.bam \
      O=OASV2_RNA_D7_7_5_S_23_bwamem.WithLB.bam \
      RGID=OASV2_RNA_D7_7_5_S_23 \
      RGLB=OASV2_RNA_D7_7_5_S_23 \
      RGPL=Illumina \
      RGPU=NA \
      RGSM=OASV2_RNA_D7_7_5_S_23         
# SCREEN ID: 9878.pts-13.pespenilab       
      
      
      
# MERGING
# Merge all the bam files into one : Screen ID: 15137.pts-13.pespenilab
/data/programs/samtools merge merged.bam *.bam 

# SORTING
# NOTE: Need to specify a different tmp directory or it will fill the rhel-root... (CURRENTLY HERE)
/data/programs/sambamba_v0.6.0 sort -m 8G -t 8 -p --tmpdir=/data/urchinlarvae2018/mapped/tmp merged.bam merged.sorted.bam
SCREEN ID: 24762.pts-0.pespenilab

-------------------------------------------------------------------------------------------------------------------------------

## IGNORE: REMOVING PCR DUPLICATES (PICARD TOOLS INSTEAD OF SAMBAMBA) ##
  
  java ­jar MarkDuplicates.jar INPUT=merged.sorted.bam
       OUTPUT=merged.sorted_dupFree.bam
       METRICS_FILE=metrics.txt CREATE_INDEX=true
       VALIDATION_STRINGENCY=LENIENT  
         
java -jar picard.jar BuildBamIndex \ 
    INPUT=dedup_reads.bam   

### HOLDING OFF ON THIS, UNCLEAR IF MARKED DUPLICATES WILL BE PICKED UP BY SAMTOOLS MPILEUP. GOING TO USE SAMTOOLS RMDUP INSTEAD: Ebert et al., 2016: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4965708/

-------------------------------------------------------------------------------------------------------------------------------

# REMOVING PCR DUPLICATES WITH samtools rmdup
Usage:  samtools rmdup [-sS] <input.srt.bam> <output.bam>

/data/programs/samtools rmdup -s merged.sorted.bam merged.sorted.rmdup.bam

# GOT AN ERROR, ISN'T WORKING. GOING TO TRY PICARD TOOLS SINCE THERE IS AN OPTION TO NOT JUST MARK DUPLICATES BUT TO REMOVE THEM (didn't realize this before)   


# REMOVING DUPLICATES WITH PICARD TOOLS
java ­jar /data/programs/picard.jar MarkDuplicates INPUT=merged.sorted.bam \
       OUTPUT=merged.sorted.rmdup.bam  REMOVE_DUPLICATES=true \
       METRICS_FILE=PCRdups.txt CREATE_INDEX=true \
       VALIDATION_STRINGENCY=SILENT 
       
       
       
  java ­jar /data/programs/picard.jar MarkDuplicates \
		I=merged.sorted.bam \
		O=merged.sorted.rmdup.bam \
		M=PCRdups.txt \
		REMOVE_DUPLICATES=true
         
  java -jar picard.jar BuildBamIndex \ 
       INPUT=dedup_reads.bam  



# COMMAND THAT WORKED FOR REMOVING DUPLICATES
java -jar /data/programs/picard.jar MarkDuplicates I=merged.sorted.bam O=merged.sorted.rmdup.bam M=PCRdups.txt REMOVE_DUPLICATES=TRUE

SCREEN ID: 62046.pts-24.pespenilab


NOTE: AFTER REMOVING DUPLICATES, BAM FILE SIZE WENT FROM 9769165024 to 3299599512 (66% loss of data)

### Going to make VCF from merged.sorted.bam and merged.sorted.rmdup.bam files ###
      
      
# INDEXING
# Run samtools index (with original samtools, not version 019)
/data/programs/samtools index merged.sorted.rmdup.bam 
SCREEN ID: 5907.pts-0.pespenilab


### VCF
# Making VCF file with samtools version 1.4 # Removing skip-indels

# WITHOUT DUPLICATES
/data/programs/samtools-1.4.1/samtools mpileup -u -t DP,AD,INFO/AD -f /users/a/m/amakukho/RESEARCH/OASV2/Spur_3.1.LinearScaffold.fa merged.sorted.rmdup.bam | /data/programs/bcftools/bin/bcftools call -mv  > RNA_OASV2_urchinlarvae2018_NoDups.vcf

SCREEN ID: 7282.pts-0.pespenilab

# WITH DUPLICATES
/data/programs/samtools-1.4.1/samtools mpileup -u -t DP,AD,INFO/AD -f /users/a/m/amakukho/RESEARCH/OASV2/Spur_3.1.LinearScaffold.fa merged.sorted.bam | 
/data/programs/bcftools/bin/bcftools call -mv  > RNA_OASV2_urchinlarvae2018.vcf 

SCREEN ID: 7300.pts-0.pespenilab


------

<div id='id-section4'/> 

### Page 4:  Working with vcf files (NoDups & with dups) to get AFs & compare to OASV2 DNA (in progress)
